### PR TITLE
feat(port): Keyboard format updates, tweaks 

### PR DESCRIPTION
**Pull request by Kenan2000**
Ports these pull requests 

https://github.com/CleverRaven/Cataclysm-DDA/pull/40961,
https://github.com/CleverRaven/Cataclysm-DDA/pull/41866,
https://github.com/CleverRaven/Cataclysm-DDA/pull/47263,
https://github.com/CleverRaven/Cataclysm-DDA/pull/56212, 
https://github.com/CleverRaven/Cataclysm-DDA/pull/46796,
https://github.com/CleverRaven/Cataclysm-DDA/pull/55152,
https://github.com/CleverRaven/Cataclysm-DDA/pull/57266,

This also includes pull requests #4627, #4623 by @chaosvolt for easier mergeability. This depends on https://github.com/cataclysmbnteam/Cataclysm-BN/pull/4634 so it cannot be merged until it's merged 

`activity_handlers.cpp` file, line 1257, `crafting_gui.cpp` file, line 718 throw errors during compiling which, based on conversation with @Coolthulhu, will be fixed with help of Coolthulhu or other developers or both. `action.cpp` file, line 930 could not be ported because it throws an error so it was added to a comment with explanation 

Coauthors have been linked properly via GitHub Desktop built-in function to do this work hopefully

### CODE REVIEW

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -1253,7 +1256,7 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
         // those lines are for XP gain with dissecting. It depends on the size of the corpse, time to dissect the corpse and the amount of bionics you would gather.
         int time_to_cut = size_factor_in_time_to_cut( corpse->size );
         int level_cap = std::min<int>( MAX_SKILL, ( corpse->size + ( cbms.size() * 2 + 1 ) ) );
```
Comment (Coolthulhu) [inline src/activity_handlers.cpp:1258]:
```suggestion
        int level_cap = std::min<int>( MAX_SKILL, ( static_cast<int>( corpse->size ) + ( cbms.size() * 2 + 1 ) ) );
```

**Review by Coolthulhu — CHANGES_REQUESTED** [review]

File: src/activity_handlers.cpp (hunk 1 above)
Comment (Coolthulhu) [reply src/activity_handlers.cpp:1258]: BTW same change can be applied in other PRs - it seems that all of them failed at this line.

**Comment by chaosvolt** [comment]
Okay so evidently https://github.com/cataclysmbnteam/Cataclysm-BN/pull/4636 contained the file changes for this PR and the other one so these can be closed? According to Kenan at least.
image

I'm going to hope this is the right descision, if anyone spots anything vital in this re-open and fix merge conflicts then, bleh.

**Comment by chaosvolt** [comment]
Per request but marking it as WIP afterward.

**Comment by Kenan2000** [comment]
Closing this because pull request of @KheirFerrum includes fixed version of this work

**Pull request closed without merging**